### PR TITLE
Issue/5711 interac refund error handling

### PR DESCRIPTION
**Pull request by AnirudhBhat**
part of: #5711  

### Description

This PR handles error case scenarios for Interac refunds. Most of the Interac refund failures will be handled with a generic "refund failed" screen. The reason for that is being the Stripe terminal SDK doesn't properly communicate the error code to the app. You can find more about this in the conversation [here](https://github.com/woocommerce/woocommerce-android/pull/p1647343856332389-slack-C01G168NFC2)

### Testing instructions

Green CI should be enough. In order to see what the end product looks like, please refer to [this](https://github.com/woocommerce/woocommerce-android/pull/6157) PR.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import com.stripe.stripeterminal.external.models.TerminalException
+import com.woocommerce.android.cardreader.payments.CardInteracRefundStatus.InteracRefundFailure
+import com.woocommerce.android.cardreader.payments.CardInteracRefundStatus.RefundStatusErrorType.Cancelled
+import com.woocommerce.android.cardreader.payments.CardInteracRefundStatus.RefundStatusErrorType.DeclinedByBackendError
+import com.woocommerce.android.cardreader.payments.CardInteracRefundStatus.RefundStatusErrorType.Generic
+import com.woocommerce.android.cardreader.payments.CardInteracRefundStatus.RefundStatusErrorType.NoNetwork
+import com.woocommerce.android.cardreader.payments.RefundParams
+
+class RefundErrorMapper {
```
Comment (kidinov) [inline libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt]: Can be internal if I'm not mistaken

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            "offline_pin_required",
+            "online_or_offline_pin_required" -> DeclinedByBackendError.CardDeclined.PinRequired
+
+            "pin_try_exceeded" -> DeclinedByBackendError.CardDeclined.TooManyPinTries
+
+            "testmode_decline" -> DeclinedByBackendError.CardDeclined.TestCard
+
+            "test_mode_live_card" -> DeclinedByBackendError.CardDeclined.TestModeLiveCard
+            else -> when (exception.apiError?.code) {
+                "amount_too_small" -> DeclinedByBackendError.AmountTooSmall
```
Comment (kidinov) [inline libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt]: So you can not refund < than 50 cents? I thought this is only for a payment

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardInteracRefundStatus.kt [hunk 3]
```diff
@@ -1,12 +1,129 @@
 package com.woocommerce.android.cardreader.payments
 
-import com.stripe.stripeterminal.external.models.TerminalException
-
 sealed class CardInteracRefundStatus {
     object InitializingInteracRefund : CardInteracRefundStatus()
```
Comment (kidinov) [inline libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardInteracRefundStatus.kt:4]: Just noticed that `InitializingInteracRefund` and `WaitingForInput` are not used at the moment. The first one probably should be removed. Not sure about the second at the moment

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt (hunk 1 above)
Comment (kidinov) [reply libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt]: Btw, it looks exactly (or almost) the same as `PaymentErrorMapper`. Am I wondering can we reuse that mapper?

**Review by kidinov — COMMENTED** [review]
The code looks clean and valid! Just left a couple of points to discuss

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt (hunk 1 above)
Comment (AnirudhBhat) [reply libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt]: >Can be internal if I'm not mistaken

You are correct. https://github.com/woocommerce/woocommerce-android/pull/6138/commits/5b3ed4853ee08023da66b064e66c5c921b8cb8f4

>Btw, it looks exactly (or almost) the same as PaymentErrorMapper. Am I wondering can we reuse that mapper?

It does look almost the same. The reason I didn't abstract these 2 is at the moment these 2 look similar. However, I wanted to give it some time before abstracting and if we still feel the same, I will create a separate PR for the refactoring. WDYT?

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt (hunk 2 above)
Comment (AnirudhBhat) [reply libs/cardreader/src/main/java/com/woocommerce/android/cardreader/internal/payments/RefundErrorMapper.kt]: Actually, I just tried refunding 0.10$ and it seems to work just fine. Removed the "amount_too_small" case here https://github.com/woocommerce/woocommerce-android/pull/6138/commits/ecdac7f28b0c75e0be6b82c75b87669e087cd6bc

File: libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardInteracRefundStatus.kt (hunk 3 above)
Comment (AnirudhBhat) [reply libs/cardreader/src/main/java/com/woocommerce/android/cardreader/payments/CardInteracRefundStatus.kt:4]: Removed `WaitingForInput` here https://github.com/woocommerce/woocommerce-android/pull/6138/commits/5a6c42db1ca501ef2b98a43b564a294096ee9513. I decided to keep `InitializingInteracRefund` to have all the different states related to Interac refund in one sealed class and it is now being used here https://github.com/woocommerce/woocommerce-android/pull/6157/commits/44044b89eb60a122a207fbe85276b4bc3fe51719

**Pull request merged**